kclvm/tools/src/LSP/src/util.rs

```diff
@@ -64,13 +64,30 @@ pub(crate) struct Params {
     pub module_cache: Option<KCLModuleCache>,
     pub scope_cache: Option<KCLScopeCache>,
     pub vfs: Option<KCLVfs>,
+    pub compile_unit_cache: Option<KCLCompileUnitCache>,
 }
 
 pub(crate) fn compile_with_params(
     params: Params,
 ) -> anyhow::Result<(Program, IndexSet<Diagnostic>, GlobalState)> {
     // Lookup compile unit (kcl.mod or kcl.yaml) from the entry file.
-    let (mut files, opt) = lookup_compile_unit(&params.file, true);
+    let (mut files, opt) = match &params.compile_unit_cache {
```

Comment:
I think the KCL IDE needs to be able to perceive files such as `kcl.mod` and `kcl.yaml` even if `json` and `yaml`, as they are the part of the incremental compilation system.

Reply:
lsp will clear compile unit cache when edit  config file(`kcl.mod` or `kcl.yaml`)

add config file watcher in https://github.com/kcl-lang/vscode-kcl/pull/41
and
```
    fn on_did_change_watched_files(
        &mut self,
        params: lsp_types::DidChangeWatchedFilesParams,
    ) -> anyhow::Result<()> {
        for change in params.changes {
            let path = from_lsp::abs_path(&change.uri)?;
            self.loader.handle.invalidate(path.clone());
            if KCL_CONFIG_FILE.contains(&path.file_name().unwrap().to_str().unwrap()) {
                self.compile_unit_cache.write().clear();
            }
        }

        Ok(())
    }
```